src/middleware/error_handling.ts

```diff
@@ -7,42 +8,52 @@ import {
     ErrorBodyWithHTTPStatusCode,
     GeneralErrorCodes,
     generalErrorCodeToReason,
+    RevertAPIError,
     ValidationError,
 } from '../errors';
+import { logger } from '../logger';
 
 /**
  * Wraps an Error with a JSON human readable reason and status code.
  */
 export function generateError(err: Error): ErrorBodyWithHTTPStatusCode {
-    if ((err as any).isAPIError) {
-        const apiError = err as APIBaseError;
-        const statusCode = apiError.statusCode;
-        if (apiError.statusCode === HttpStatus.BAD_REQUEST) {
-            const badRequestError = apiError as BadRequestError;
-            if (badRequestError.generalErrorCode === GeneralErrorCodes.ValidationError) {
-                const validationError = badRequestError as ValidationError;
+    if (isAPIError(err)) {
+        const { statusCode } = err;
+        if (isBadRequestError(err)) {
+            const { generalErrorCode: code } = err;
+            if (isValidationError(err)) {
                 return {
                     statusCode,
                     errorBody: {
-                        code: badRequestError.generalErrorCode,
-                        reason: generalErrorCodeToReason[badRequestError.generalErrorCode],
-                        validationErrors: validationError.validationErrors,
+                        code,
+                        reason: generalErrorCodeToReason[code],
+                        validationErrors: err.validationErrors,
+                    },
+                };
+            } else if (isRevertAPIError(err)) {
+                return {
+                    statusCode,
+                    errorBody: {
+                        code,
+                        reason: err.name,
+                        values: err.values,
                     },
                 };
             } else {
                 return {
                     statusCode,
                     errorBody: {
-                        code: badRequestError.generalErrorCode,
-                        reason: generalErrorCodeToReason[badRequestError.generalErrorCode],
+                        code,
+                        reason: generalErrorCodeToReason[code],
                     },
                 };
             }
         } else {
+            logger.info({ isAPIError: true, message: err.message });
```

Comment:
I added similar logging in #29. We can search for errors in logs using `res.statusCode`. So this might be unnecessarily noisy?
![Screen Shot 2019-12-13 at 9 52 27 AM](https://user-images.githubusercontent.com/8582774/70820763-4ba6b700-1d8e-11ea-934c-3fbaafa65cb9.png)